File: benchmarks/reference/dict_access.py
import time

def main():
    start = time.time()
    d = {}
    for i in range(10000):
        d["key" + str(i)] = i
    
    total_sum = 0
    for i in range(10000):
        total_sum += d["key" + str(i)]
    
    print(f"Dict Sum: {total_sum}")
    elapsed = time.time() - start
    print(f"Time: {elapsed:.6f}")

if __name__ == "__main__":
    main()

File: benchmarks/reference/dict_access.java
import java.util.HashMap;

public class dict_access {
    public static void main(String[] args) {
        long start = System.nanoTime();
        HashMap<String, Integer> d = new HashMap<>();
        for (int i = 0; i < 10000; i++) d.put("key" + i, i);
        
        long totalSum = 0;
        for (int i = 0; i < 10000; i++) totalSum += d.get("key" + i);
        
        System.out.println("Dict Sum: " + totalSum);
        double elapsed = (System.nanoTime() - start) / 1e9;
        System.out.println("Time: " + elapsed);
    }
}
